composite-actions/lossprevention/acceptance/action.yml

```diff
@@ -46,10 +46,22 @@ runs:
         service-account: basketauditfrontend@${{ steps.gcloud.outputs.project-id }}.iam.gserviceaccount.com
         audiences: basketauditfrontend
 
+    - name: List collection files
+      id: list_collections
+      run: |
+        collections=$(find "${{ inputs.collections_folder }}" -name '*.postman_collection.json' -type f)
+        echo "Found collections:"
+        echo "$collections"
+        echo "::set-output name=collections::$(echo "$collections" | tr '\n' ',')"
+
     - name: Acceptance tests
       shell: bash
       run: |
           npm install newman --no-save --silent
-          npx newman run ${{ inputs.collection }} \
-            --environment ${{ inputs.environment }} \
-            --env-var iamToken=$IDENTITY_TOKEN
+          IFS=',' read -ra COLLECTIONS_ARRAY <<< "${{ steps.list_collections.outputs.collections }}"
+          for collection in "${COLLECTIONS_ARRAY[@]}"; do
+            echo "Running tests for collection: $collection"
+            npx newman run "$collection" \
+              --environment tests/integration_test.staging.postman_environment.json \
```

Comment:
Do we really need step to list entire collection. Can we reduce the number of unnecessary echo statements?

Reply:
Also may be this is not running the collections in parallel. Is there way to make it run in parallel?

Reply:
Steps and echo statements reduced

Reply:
Discussed on the parallel approach